### PR TITLE
RFC - Improving deploying Dask docs

### PR DESCRIPTION
**Pull request by scharlottej13**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

RFC PR for making it easier for those looking for Dask deployment options to find available options (including Coiled). ~This includes a draft for a new docs page on deployment best practices, based on @mrocklin's [PyData NYC 2022 talk](https://nyc2022.pydata.org/cfp/talk/M38FJZ/).~

~This deployment best practices page is intended as a hub for other relevant Dask projects (i.e. Dask Cloud Provider, Dask Kubernetes, and Dask Gateway) to link back to. This could be a sentence at the top of the docs index for these projects. Something like "Dask Gateway is one of many ways to deploy Dask; for an overview of other options see the Dask documentation on Deploying Best Practices." We can also link to this page from Dask examples that mention deployment options (e.g. https://examples.dask.org/applications/embarrassingly-parallel.html#Start-Dask-Client-for-Dashboard)~

Edit - "Deployment Considerations" page was added in #9933.

This PR also includes other suggestions of where to link to Coiled, where appropriate.

cc @jacobtomlinson @dchudz @gjoseph92

### CODE REVIEW

File: docs/source/deploying-best-practices.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+Deploying Best Practices
```
Comment (mrocklin) [inline docs/source/deploying-best-practices.rst:1]: This doesn't seem as much like a normal best practices doc as it does a general blog post.  It's more of a narrative style.  

I think it makes sense as a blog post and could be good, but I don't think it makes sense here as it is.

File: docs/source/deploying-cloud.rst [hunk 2]
```diff
@@ -14,9 +14,9 @@ You may want to consider the following options:
     and `Dask-Yarn <https://yarn.dask.org>`_.
 
     Specific documentation for the popular Amazon EMR service can be found
-    `here <https://yarn.dask.org/en/latest/aws-emr.html>`_
+    `here <https://yarn.dask.org/en/latest/aws-emr.html>`_.
 3.  Directly launching cloud resources such as VMs or containers via a cluster manager with
-    `Dask Cloud Provider <https://cloudprovider.dask.org/en/latest/>`_
+    `Dask Cloud Provider <https://cloudprovider.dask.org/en/latest/>`_ or `Coiled <https://www.coiled.io/>`_.
```
Comment (mrocklin) [inline docs/source/deploying-cloud.rst:19]: From a users perspective I think that these two are pretty different.  It's true that under the hood they both use VMs, but I think that commercial or managed services might make more sense as a separate bullet point.

File: docs/source/ecosystem.rst [hunk 3]
```diff
@@ -69,6 +69,7 @@ There are many different implementations of the Dask distributed cluster.
   installations.
 - `dask-cloudprovider <https://cloudprovider.dask.org>`_: Deploy Dask on various cloud platforms such as AWS, Azure, and GCP
   leveraging cloud native APIs.
+- `Coiled <https://www.coiled.io/>`_: Deploy Dask in the cloud from your AWS or Google Cloud accounts.
```
Comment (mrocklin) [inline docs/source/ecosystem.rst:72]: I think that we want to call this out as a managed/commercial offering in this description. That's the first major distinction from Dask cloud provider

File: docs/source/how-to/manage-environments.rst [hunk 4]
```diff
@@ -62,6 +62,12 @@ in-place, include:
 - cluster install method (e.g., `parcels`_): depending on your infrastructure, there may be
   ways to install specific binaries to all workers in a cluster.
 
+There are also tools that replicate your local machine's environment in the cloud. If you are
+using `Coiled <https://www.coiled.io/>`_, for example, your local Python environment is automatically
+replicated in your Dask cluster in the cloud (see the `Coiled documentation on package sync <https://docs.coiled.io/user_guide/package_sync.html>`_).
```
Comment (mrocklin) [inline docs/source/how-to/manage-environments.rst:67]: I like this with my Coiled hat on, I'm not sure how I feel with my community hat.  I'd defer to Jacob on this one.

**Review by mrocklin — CHANGES_REQUESTED** [review]
Thanks for writing this up Sarah!  Some of this seems good to me, some not.  I've put some remarks here but mostly I think we leave this and wait and see what other non Coiled affiliated people think.

File: docs/source/deploying-best-practices.rst (hunk 1 above)
Comment (mrocklin) [reply docs/source/deploying-best-practices.rst:1]: Also, not that it's super relevant here, but the actual talk was released a couple of weeks ago (albeit with a slight visual lag)

https://www.youtube.com/watch?v=5hUkUj1VYW4

File: docs/source/deploying-cloud.rst (hunk 2 above)
Comment (gjoseph92) [reply docs/source/deploying-cloud.rst:19]: I came here to see if we'd added Coiled to this list. Agreed with Matt that Coiled should be a separate bullet, but I think it definitely makes sense to have on the list.

```suggestion
    `Dask Cloud Provider <https://cloudprovider.dask.org/en/latest/>`_.
4. `Coiled <https://www.coiled.io/>`_, a commercial service that manages launching Dask clusters on your own cloud infrastructure.
```

File: docs/source/deploying-best-practices.rst (hunk 1 above)
Comment (gjoseph92) [reply docs/source/deploying-best-practices.rst:1]: I really like the content in this talk. I think this would be valuable to have on a page in the docs (in a less blog-post style)—basically "all the other stuff you have to deal with when deploying dask that you probably haven't thought of yet". Giving people a warning of all the problems they're likely to run into, instead of having to learn the hard way, is very valuable.

I don't think I'd title it "Deploying Best Practices" though. Maybe "Considerations when deploying Dask"?

**Review by gjoseph92 — COMMENTED** [review]
> page is intended as a hub for other relevant Dask projects (i.e. Dask Cloud Provider, Dask Kubernetes, and Dask Gateway) to link back to. This could be a sentence at the top of the docs index for these projects. Something like "Dask Gateway is one of many ways to deploy Dask; for an overview of other options see the Dask documentation

Such a page sounds a lot like https://docs.dask.org/en/stable/deploying.html to me. That already looks like a "hub" page that lists and links to the many options for deploying Dask.

It's a dense list right now, and we could probably do a lot to make it easier to navigate:
- cut down and summarize the current prose into ~1 paragraph
- organize the list of links by where you're deploying dask to (single machine, HPC, k8s, cloud, ad-hoc cluster
- perhaps also organize the list (via tabs at the top?) by who you're deploying dask for: "just me" vs "my team"? For example, dask-gateway likely isn't relevant to a single user; dask-cloudprovider is probably not a good solution for a team. Some options, including the commercial solutions, would be in both.

So basically, I like the idea of having a good "hub" page that makes it easy for users to navigate all the different options for deploying Dask. I just think the title of that page is probably "Deploy Dask Clusters", not "Dask Deployment Best Practices". Since "Deploy Dask Clusters" already exists, maybe we should be improving that page, not making a new one?

**Comment by gjoseph92** [comment]
Also, in the TOC, it might be worth adding another page for "Commercial services", "Managed services", "Dask-as-a-Service", something like that.

Screen Shot 2023-02-04 at 12 57 54 PM

**Comment by gjoseph92** [comment]
Also, in the "For IT" section of the FAQ https://docs.dask.org/en/stable/faq.html#how-would-i-set-up-dask-on-institutional-hardware, it would make a lot of sense to mention commercial services. (Also in the "How do I manage users" section.)

File: docs/source/how-to/manage-environments.rst (hunk 4 above)
Comment (jacobtomlinson) [reply docs/source/how-to/manage-environments.rst:67]: I'm fine with this. It is a statement of fact and is mentioned in a footnote style below what I would consider to be the "Dask Community" bit.

File: docs/source/deploying-best-practices.rst (hunk 1 above)
Comment (scharlottej13) [reply docs/source/deploying-best-practices.rst]: This page has been moved to https://github.com/dask/dask/pull/9933

File: docs/source/deploying.rst [hunk 5]
```diff
@@ -93,62 +92,83 @@ Those workers are then allocated physical hardware resources.
...
-   An overview of cluster management with Dask distributed.
-
 .. _deployment-options:
 
-To summarize, you can use the default, single-machine scheduler to use Dask
-on your local machine. If you'd like use a cluster *or* simply take advantage
-of the :doc:`extensive diagnostics <../diagnostics-distributed>`,
-you can use Dask distributed. The following resources explain
-in more detail how to set up Dask on a variety of local and distributed hardware:
+The following resources explain how to set up Dask on a variety of local and distributed hardware.
```
Comment (scharlottej13) [inline docs/source/deploying.rst:98]: @gjoseph92 I made a first pass at your suggestions in https://github.com/dask/dask/pull/9912#pullrequestreview-1284106497 but not sure I have the categories quite right.

I also tried a couple tab variations, I think the list seemed a bit long with tabs for individual/team/IT, but maybe we can condense it?
Screen Shot 2023-02-15 at 6 47 34 PM

Comment (scharlottej13) [reply docs/source/deploying.rst:98]: I also tried nested tabs, but those seem a bit busy:
Screen Shot 2023-02-15 at 6 11 52 PM

File: docs/source/deploying.rst [hunk 6]
```diff
@@ -93,62 +92,83 @@ Those workers are then allocated physical hardware resources.
...
+Cloud
+~~~~~
+            
+See :doc:`deploying-cloud` for more details.
+
+- `Dask-Yarn <https://yarn.dask.org>`_
+    Deploy Dask on YARN clusters, such as are found in traditional Hadoop installations.
+- `Dask Cloud Provider <https://cloudprovider.dask.org/en/latest/>`_
+    Constructing and managing ephemeral Dask clusters on AWS, DigitalOcean, GCP, Azure, and Hertzner
+
```
Comment (gjoseph92) [inline docs/source/deploying.rst:152]: I kinda want to see Coiled here?

File: docs/source/how-to/manage-environments.rst [hunk 7]
```diff
@@ -62,6 +62,7 @@ in-place, include:
 - cluster install method (e.g., `parcels`_): depending on your infrastructure, there may be
   ways to install specific binaries to all workers in a cluster.
 
+.. _Coiled: https://www.coiled.io/
```
Comment (gjoseph92) [inline docs/source/how-to/manage-environments.rst:65]: Where is this link used?

File: docs/source/deploying.rst (hunk 5 above)
Comment (gjoseph92) [reply docs/source/deploying.rst:98]: I think it would only be individuals / teams; not sure what things would go under "For IT".

I do still like the idea of this, but I agree that I'm not sure how to organize it. Probably the first option (just one set of tabs)?

File: docs/source/faq.rst [hunk 8]
```diff
@@ -184,6 +184,16 @@ those on Hadoop, HPC, Kubernetes, and Cloud clusters.
 
     For more information see :doc:`deploying-kubernetes`
 
+4. **Managed cluster solutions**: These commercial options often handle creation and management of
```
Comment (gjoseph92) [inline docs/source/faq.rst:187]: I'm not familiar with which of these (besides Coiled) actually support deploying on your own resources (your own cloud account / k8s). We should probably check on that.

**Review by gjoseph92 — COMMENTED** [review]
This looks good. I think there's more work we can do to the deployment page (I still find the text a bit long, and having the individual vs team split would be nice), but I think we should tackle that incrementally. This is already an improvement to the organization!

File: docs/source/ecosystem.rst (hunk 3 above)
Comment (scharlottej13) [reply docs/source/ecosystem.rst]: Sounds good, I've added a `Commercial Dask Deployment Options` section at the bottom with Coiled listed as one option.

File: docs/source/how-to/manage-environments.rst (hunk 4 above)
Comment (scharlottej13) [reply docs/source/how-to/manage-environments.rst]: I actually decided to remove this section, @gjoseph92 added it to the Deployment Considerations page in #9933 and I think it's a clearer fit there

File: docs/source/faq.rst (hunk 8 above)
Comment (scharlottej13) [reply docs/source/faq.rst]: Ah good point. I changed this to simply `**Commercial Dask deployment:**`

File: docs/source/how-to/manage-environments.rst (hunk 7 above)
Comment (scharlottej13) [reply docs/source/how-to/manage-environments.rst]: thanks for catching this, removed it!

File: docs/source/deploying.rst (hunk 5 above)
Comment (scharlottej13) [reply docs/source/deploying.rst:98]: Thanks @gjoseph92! I left it as-is (without tabs) for now, but happy to revisit this.

File: docs/source/deploying.rst (hunk 6 above)
Comment (scharlottej13) [reply docs/source/deploying.rst:155]: Added it to the list!

File: docs/source/deploying.rst [hunk 9]
```diff
@@ -93,62 +92,83 @@ Those workers are then allocated physical hardware resources.
...
+- `Dask-MPI <http://mpi.dask.org/en/latest/>`_
+    Deploy Dask from within an existing MPI environment.
+- `Dask Gateway <https://gateway.dask.org/usage.html>`_
+
+Kubernetes
+~~~~~~~~~~
+
+See :doc:`deploying-kubernetes` for more details.
+
+- :doc:`Helm <deploying-kubernetes-helm>`
```
[the remark this answers is not in the document]
Comment (jacobtomlinson) [reply docs/source/deploying.rst:138]: Sure it's a fine option for beginners, although it probably isn't the first choice now that we have the operator.

**Review by jacobtomlinson — APPROVED** [review]
This looks great, thanks for the effort here.
